### PR TITLE
AB#67046 Added check answers

### PR DESCRIPTION
**Pull request by rafalgajewski**

### CODE REVIEW

File: src/HE.Investment.AHP.Contract/FinancialDetails/FinancialDetails.cs [hunk 1]
```diff
@@ -47,19 +47,29 @@ public class FinancialDetails
...
-                + OtherCapitalContributions ?? 0
-                + SharedOwnershipSalesContribution ?? 0
-                + TransferValueOfHomes ?? 0;
+            result += (PurchasePrice ?? 0) + (ExpectedWorkCost ?? 0) + (ExpectedOnCost ?? 0);
+
+            return result;
+        }
+    }
+
+    public decimal TotalExpectedContributions
```
Comment (Antek012) [inline src/HE.Investment.AHP.Contract/FinancialDetails/FinancialDetails.cs:61]: Having calculations in contract class is not a good idea. Please move this to domain.

File: src/HE.Investment.AHP.Domain/FinancialDetails/Constants/FinancialDetailsValidationFieldNames.cs [hunk 2]
```diff
@@ -45,4 +44,7 @@ public static class FinancialDetailsValidationFieldNames
     public const string LotteryGrants = nameof(LotteryGrants);
 
     public const string OtherPublicBodiesGrants = nameof(OtherPublicBodiesGrants);
+
+    // Check answers
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/FinancialDetails/Constants/FinancialDetailsValidationFieldNames.cs:48]: What does this comment mean?

File: src/HE.Investment.AHP.WWW/Controllers/FinancialDetailsController.cs [hunk 3]
```diff
@@ -236,7 +248,40 @@ public async Task<IActionResult> Grants(Guid applicationId, FinancialDetailsGran
...
+    public async Task<IActionResult> CheckAnswers(Guid applicationId, CancellationToken cancellationToken)
+    {
+        var model = await _financialDetailsSummaryModelFactory.GetFinancialDetailsAndCreateSummary(Url, applicationId, cancellationToken);
+
+        return View(model);
+    }
+
+    [HttpPost("check-answers")]
+    [WorkflowState(FinancialDetailsWorkflowState.CheckAnswers)]
+    public async Task<IActionResult> CheckAnswers(Guid applicationId, FinancialDetailsCheckAnswersModel model, CancellationToken cancellationToken)
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Controllers/FinancialDetailsController.cs:265]: I think this method should be named "Complete"

File: src/HE.Investment.AHP.WWW/Models/FinancialDetails/Factories/FinancialDetailsSummaryModelFactory.cs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+using HE.Investments.Common.Extensions;
+using HE.Investments.Common.WWW.Components.SectionSummary;
+using HE.Investments.Common.WWW.Utils;
+using HE.Investments.Loans.Common.Utils.Constants.FormOption;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace HE.Investment.AHP.WWW.Models.FinancialDetails.Factories;
+
+public class FinancialDetailsSummaryModelFactory : IFinancialDetailsSummaryModelFactory
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Models/FinancialDetails/Factories/FinancialDetailsSummaryModelFactory.cs:20]: You have to inject this class to ApplicationSummaryViewModelFactory and display FinancielDetails summary on Application CheckAnswers page.

File: src/HE.Investment.AHP.Domain/FinancialDetails/ValueObjects/PurchasePrice.cs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            if (!decimal.TryParse(actualPurchasePrice, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture, out var price))
+            {
+                OperationResult.New()
+                    .AddValidationError(FinancialDetailsValidationFieldNames.PurchasePrice, FinancialDetailsValidationErrors.InvalidActualPurchasePrice)
+                    .CheckErrors();
+            }
+
+            ActualPrice = price;
+        }
+        else
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/FinancialDetails/ValueObjects/PurchasePrice.cs:25]: I think ele(s) are not needed in this metod.

File: src/HE.Investment.AHP.Domain/FinancialDetails/ValueObjects/Grants.cs [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    {
+        var operationResult = OperationResult.New();
+
+        CountyCouncil = NumericValidator
+            .For(countyCouncil, FinancialDetailsValidationFieldNames.CountyCouncilGrants, operationResult)
+            .IsWholeNumber(FinancialDetailsValidationErrors.GenericAmountValidationError)
+            .IsBetween(0, 999999999, FinancialDetailsValidationErrors.GenericAmountValidationError)
+            .IsConditionallyRequired(!allowNulls);
+
+        DHSCExtraCare = NumericValidator
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/FinancialDetails/ValueObjects/Grants.cs:105]: DRY - you can create private method and remove all duplications.

File: src/HE.Investment.AHP.Domain/FinancialDetails/ValueObjects/Contributions.cs [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+        string? ownResources,
+        string? rCGFContributions,
+        string? otherCapitalSources,
+        string? sharedOwnershipSales,
+        string? homesTransferValue,
+        bool allowNulls = true)
+    {
+        var operationResult = OperationResult.New();
+
+        RentalIncomeBorrowing = NumericValidator
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/FinancialDetails/ValueObjects/Contributions.cs:107]: DRY - create private method

File: src/HE.Investment.AHP.Domain/FinancialDetails/Constants/FinancialDetailsValidationFieldNames.cs [hunk 8]
```diff
@@ -1,10 +1,9 @@
 namespace HE.Investment.AHP.Domain.FinancialDetails.Constants;
 public static class FinancialDetailsValidationFieldNames
 {
-    public const string ActualPurchasePrice = nameof(ActualPurchasePrice);
-
-    public const string ExpectedPurchasePrice = nameof(ExpectedPurchasePrice);
+    public const string PurchasePrice = nameof(PurchasePrice);
 
+    // Land value
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/FinancialDetails/Constants/FinancialDetailsValidationFieldNames.cs:6]: This comment is wrong - please remove.

File: src/HE.Investment.AHP.WWW/Controllers/FinancialDetailsController.cs (hunk 3 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.WWW/Controllers/FinancialDetailsController.cs:265]: earlier we agreed with @Carq  that methods should have the same names as page titles

File: src/HE.Investment.AHP.Contract/FinancialDetails/FinancialDetails.cs (hunk 1 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Contract/FinancialDetails/FinancialDetails.cs:61]: Moved.

File: src/HE.Investment.AHP.Domain/FinancialDetails/Constants/FinancialDetailsValidationFieldNames.cs (hunk 2 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/FinancialDetails/Constants/FinancialDetailsValidationFieldNames.cs:48]: It was to make navigation on the file easier

File: src/HE.Investment.AHP.Domain/FinancialDetails/ValueObjects/PurchasePrice.cs (hunk 5 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/FinancialDetails/ValueObjects/PurchasePrice.cs:25]: Removed but it depends if we want to have ability to set value to null when user clears the filed and saves chnages

File: src/HE.Investment.AHP.Domain/FinancialDetails/ValueObjects/Contributions.cs (hunk 7 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/FinancialDetails/ValueObjects/Contributions.cs:107]: As above

File: src/HE.Investment.AHP.Domain/FinancialDetails/Constants/FinancialDetailsValidationFieldNames.cs (hunk 8 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/FinancialDetails/Constants/FinancialDetailsValidationFieldNames.cs:6]: done.

File: src/HE.Investment.AHP.WWW/Controllers/FinancialDetailsController.cs (hunk 3 above)
Comment (Antek012) [reply src/HE.Investment.AHP.WWW/Controllers/FinancialDetailsController.cs:266]: Ok. It may sense for redirect.

Comment (rafalgajewski) [reply src/HE.Investment.AHP.WWW/Controllers/FinancialDetailsController.cs]: I've changed method name as suggested

File: src/HE.Investment.AHP.WWW/Models/FinancialDetails/Factories/FinancialDetailsSummaryModelFactory.cs (hunk 4 above)
Comment (rafalgajewski) [reply src/HE.Investment.AHP.WWW/Models/FinancialDetails/Factories/FinancialDetailsSummaryModelFactory.cs]: I've added summary in the same form as on financial details check answers page.

File: src/HE.Investments.Common.WWW/Components/SectionSummary/SectionSummary.cshtml [hunk 9]
```diff
@@ -1,7 +1,13 @@
 @using HE.Investments.Common.WWW.Components.SectionSummary;
 
-@model (string title, IList<SectionSummaryItemModel> items)
+@model (string title, IList<SectionSummaryItemModel> items, string validationError)
 <gds-h2>@Model.title</gds-h2>
+@if (!string.IsNullOrWhiteSpace(Model.validationError))
```
[the remark this answers is not in the document]
Comment (rafalgajewski) [reply src/HE.Investments.Common.WWW/Components/SectionSummary/SectionSummary.cshtml]: I've used Check error component

File: src/HE.Investment.AHP.Domain/FinancialDetails/Entities/FinancialDetailsEntity.cs [hunk 10]
```diff
@@ -9,161 +14,139 @@ public FinancialDetailsEntity(ApplicationId applicationId, string applicationNam
...
+        (Contributions.SalesOfHomesOnThisScheme ?? 0) +
+        (Contributions.SalesOfHomesOnOtherSchemes ?? 0) +
+        (Contributions.OwnResources ?? 0) +
+        (Contributions.RCGFContributions ?? 0) +
+        (Contributions.OtherCapitalSources ?? 0) +
+        (Contributions.SharedOwnershipSales ?? 0) +
+        (Contributions.HomesTransferValue ?? 0);
 
-    public SalesOfHomesOnOtherSchemes? SalesOfHomesOnOtherSchemes { get; private set; }
+    public decimal TotalGrants =>
```
[the remark this answers is not in the document]
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/FinancialDetails/Entities/FinancialDetailsEntity.cs]: Done

File: src/HE.Investment.AHP.Domain/FinancialDetails/Entities/FinancialDetailsEntity.cs [hunk 11]
```diff
@@ -9,161 +14,139 @@ public FinancialDetailsEntity(ApplicationId applicationId, string applicationNam
...
+        }
     }
 
-    public void ProvideExpectedCosts(ExpectedWorksCosts? expectedWorksCosts, ExpectedOnCosts? expectedOnCosts)
+    public void ProvideContributions(Contributions contributions)
     {
-        ExpectedWorksCosts = expectedWorksCosts;
-        ExpectedOnCosts = expectedOnCosts;
+        Contributions = contributions;
+        if (Contributions.RentalIncomeBorrowing != null
```
[the remark this answers is not in the document]
Comment (rafalgajewski) [reply src/HE.Investment.AHP.Domain/FinancialDetails/Entities/FinancialDetailsEntity.cs]: Done

**Review by Carq — APPROVED** [review]

**Pull request merged**